### PR TITLE
feature: #9 | Add the spotless plugin and refactor the code

### PR DESCRIPTION
**Pull request by astha-singh-2445**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 ARG PUBLIC_KEY
 ARG KAFKA_BOOTSTRAP_SERVERS
 WORKDIR /app
 COPY gradle gradle
 COPY src src
 COPY gradlew gradlew
-COPY build.gradle build.gradle
-COPY settings.gradle settings.gradle
+COPY build.gradle.kts build.gradle.kts
+COPY settings.gradle.kts settings.gradle
```
Comment (ShivamNagpal) [inline Dockerfile:10]: This instruction should be
`COPY settings.gradle.kts settings.gradle.kts`

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	implementation("org.springdoc:springdoc-openapi-ui:1.7.0")
+
+	testImplementation("org.springframework.boot:spring-boot-starter-test")
+	testImplementation("io.projectreactor:reactor-test")
+	testImplementation("org.springframework.kafka:spring-kafka-test")
+}
+
+tasks.withType<Test> {
+	useJUnitPlatform()
+}
```
Comment (ShivamNagpal) [inline build.gradle.kts:62]: nit: Missing end line

File: src/main/java/com/singh/astha/notificationservice/AppConfiguration.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 @Configuration
 public class AppConfiguration {
 
-    private static final String[] SCOPES = {Constants.MESSAGING_SCOPE};
+    private static final String[] SCOPES = { Constants.MESSAGING_SCOPE };
```
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/notificationservice/AppConfiguration.java:24]: nit: The type can be changed to List<String> and the list can be initialised using List.of() and then the list can be passed directory to the GoogleCredentials.

File: src/main/java/com/singh/astha/notificationservice/config/JwtAuthenticationHandler.java [hunk 4]
```diff
@@ -27,30 +28,43 @@ public JwtAuthenticationHandler(JwtService jwtService) {
...
             );
             return;
         }
-        writeResponse(response, HttpStatus.FORBIDDEN.value(),
-                ResponseWrapper.failure(null, ErrorMessages.ACCESS_DENIED));
+        writeResponse(
+                response,
+                HttpStatus.FORBIDDEN.value(),
+                ResponseWrapper.failure(null, ErrorMessages.ACCESS_DENIED)
+        );
```
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/notificationservice/config/JwtAuthenticationHandler.java:60]: The assumption of 403 here is not correct. But as this is a Spotless PR, create a new issue to fix this in the future.

File: src/main/java/com/singh/astha/notificationservice/exceptions/handler/ResponseExceptionHandler.java [hunk 5]
```diff
@@ -7,15 +7,18 @@
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 
-
 @ControllerAdvice
 public class ResponseExceptionHandler {
```
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/notificationservice/exceptions/handler/ResponseExceptionHandler.java:11]: Let's create an issue to combine all the Exception Handlers, I don't see any advantage of keeping them separate

File: src/main/java/com/singh/astha/notificationservice/config/JwtAuthenticationHandler.java (hunk 4 above)
Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/notificationservice/config/JwtAuthenticationHandler.java:60]: Created a separate Issue for this --> https://github.com/astha-singh-2445/Notification_Service_Spring_App/issues/19

Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/notificationservice/config/JwtAuthenticationHandler.java:60]: Created Issue-> https://github.com/astha-singh-2445/Notification_Service_Spring_App/issues/20

File: Dockerfile (hunk 1 above)
Comment (astha-singh-2445) [reply Dockerfile:10]: fixed

File: build.gradle.kts (hunk 2 above)
Comment (astha-singh-2445) [reply build.gradle.kts:62]: fixed

**Review by ShivamNagpal — APPROVED** [review]

**Pull request merged**